source/draw/gpu/opengl/backend/gles3/Gles3VertexArrayObject.ooc

```diff
@@ -37,36 +37,70 @@ Gles3VertexArrayObject: class extends GLVertexArrayObject {
 				packedArray[attributeCount * dimensions * i + j + dimensions] = textureCoordinates[dimensions * i + j]
 			}
 		}
+		this _generate(packedArray[0]&, 2, vertexCount)
+	}
+	init: func ~twoDimensions (positions, textureCoordinates: FloatPoint2D[]) {
+		vertexCount := positions length
+		this _vertexCount = vertexCount
+		floatsPerVertex := 4
+		packedArray: Float[floatsPerVertex * vertexCount]
+		for (i in 0 .. vertexCount) {
+			//Positions
+			packedArray[floatsPerVertex * i + 0] = positions[i] x
+			packedArray[floatsPerVertex * i + 1] = positions[i] y
+			//Texture coordinates
+			packedArray[floatsPerVertex * i + 2] = textureCoordinates[i] x
+			packedArray[floatsPerVertex * i + 3] = textureCoordinates[i] y
+		}
+		this _generate(packedArray[0]&, 2, vertexCount)
+	}
+	init: func ~threeDimensions (positions: FloatPoint3D[], textureCoordinates: FloatPoint2D[]) {
+		vertexCount := positions length
+		this _vertexCount = vertexCount
+		floatsPerVertex := 5
+		packedArray: Float[vertexCount * floatsPerVertex]
+		for (i in 0 .. vertexCount) {
+			//Positions
+			packedArray[floatsPerVertex * i + 0] = positions[i] x
+			packedArray[floatsPerVertex * i + 1] = positions[i] y
+			packedArray[floatsPerVertex * i + 2] = positions[i] z
+			//Texture coordinates
+			packedArray[floatsPerVertex * i + 3] = textureCoordinates[i] x
+			packedArray[floatsPerVertex * i + 4] = textureCoordinates[i] y
+		}
+		this _generate(packedArray[0]&, 3, vertexCount)
```

Comment:
I know... I will remove the first constructor (the one with the C pointers) but I need an index operator for the FloatPoint classes to replace all this with a single loop (i.e. point[0] == point x). What do you think?